### PR TITLE
Appimage

### PR DESCRIPTION
**Pull request by jberkenbilt**
@KurtPfeifle This is my adaptation of your AppImage submission. I'm planning on accepting your changes with my modifications and releasing 8.0.0 tomorrow.

### CODE REVIEW

**Comment by jberkenbilt** [comment]
@KurtPfeifle My changes from your submission are as follows:
* Spacing and stylistic changes in the code enough to make it comfortable for me to work with but not enough to destroy your style (I hope)
* Removal of `-u` as a short option for `--usage`
* Documentation cleanup, including removal of the note from README.md that this was a temporary fork
* Accept any Ubuntu 14.04 release, not just 14.04.5
* Generate the PNG files dynamically from the SVG in 64, 124, 256, and 512 widths with scaled heights and white backgrounds
* Fix build-appimage to work properly when run from my docker image and not just in travis
* Remove the upload.sh part of .travis.yml. I'm not ready to include that right now.

If you've got a few minutes, give it a peek. I'm not sure why I can't see the travis CI build log for #185.

I read through your remaining comments on #181 and have either accepted or improved upon them.

I'm trying to get 8.0.0 out tomorrow, Sunday 2/25.

File: appimage/build-appimage [hunk 1]
```diff
@@ -46,14 +52,18 @@ _osversion=$(cat /etc/os-release | grep PRETTY_NAME | awk -F'=' '{print $2}' | s
 if [ x"$_osversion" != x"Ubuntu 14.04.5 LTS" ] ; then
```
Comment (KurtPfeifle) [inline appimage/build-appimage]: In line with change at https://github.com/qpdf/qpdf/pull/185/commits/568f518feb9c48b557d07923432c5fcd5b79b7c4#diff-5af93be82eec81e6f39b5d6084d6c421R52 (*"Being more general about the environment"*) this should probably also be changed to 

    if [[ ! $_osversion =~ Ubuntu\ 14.04.*\ LTS ]]; then

Comment (KurtPfeifle) [reply appimage/build-appimage]: ...or is it just me being confused about *were* did occur *which* change in the sequence of commits?

**Comment by jberkenbilt** [comment]
When I ran build-appimage, it didn't generate a zsync file. What generates that?

**Comment by jberkenbilt** [comment]
I'm going to do a mini 8.0.rc1 release with just source tarball and the appimage. I'll let you know when it's up...probably a few minutes. Then I'll be offline for a few hours but will want to push out the release when I'm back online later this afternoon. I'm not considering fixing the updateinfo problem or the zsync problem essential for this release. There will be more releases.

**Comment by jberkenbilt** [comment]
@KurtPfeifle 8.0.rc1 is released using my full release process except that I didn't bother with Windows binaries. Please give it a look. If this is good, I will update the version numbers and changelog and call it 8.0.0 later today. I'm offline for the next several hours.

**Comment by jberkenbilt** [comment]
Change are in master, but I can still obviously add commits, so we can still use this PR for discussion until 8.0.0.

**Comment by KurtPfeifle** [comment]
@jberkenbilt wrote:

> *"When I ran build-appimage, it didn't generate a zsync file. What generates that?"*

The line which finally generates the AppImage also can generate the .zsync file... 

I don't see your branch *`jberkenbilt:appimage`*, but in master it is this (from https://github.com/qpdf/qpdf/blob/master/appimage/build-appimage#L194):

https://github.com/qpdf/qpdf/blob/a16d703f4d6df5d83cab2099e6216b50aa8fdf3a/appimage/build-appimage#L194

    PATH=./squashfs-root/usr/bin:$PATH ./squashfs-root/usr/bin/appimagetool $sign -g $appimagetool_param appdir qpdf-$VERSION-x86_64.AppImage

Specifically, here it is the simple *`-g`* param (for a *'guess'* of the updateinfo string). However, a guess only works for builds ***inside*** the Travis environment, because the tool's guesswork is based on the presence of specific *`TRAVIS_<somethings>`* environment variables.

If building ***outside of Travis*** this *`'-g'`* should be replaced by a specific and the explicit parameter:

 *`-u "zsync|http://server.domain/path/Application-latest-x86_64.AppImage.zsync"`*

Of course, replace the *'http://server.domain/path/'* part by the specific URI (http or https) where the download file(s) will be hosted. The *`'Application-latest-x86_64.AppImage'`* should be the exact name of your AppImage. The *`'.zsync'`* extension should be added into the *`'-u "...."'`* string.

The *`appimagetool`* utility should then...

1. ...print messages to the effect of "found a valid updateinfo string" while it produces the AppImage;
2. ...generate two different files,
    + *`'Application-latest-x86_64.AppImage'`* as well as
    + *`'Application-latest-x86_64.AppImage.zsync'`*
    and both files should later be hosted side by side on *`'http://server.domain/path/'`*.

Whenever an AppImage client side tool wants to check for an update for the original AppImage (which has that string embedded), it will test if there is a new AppImage version available at *`'http://server.domain/path/'`* and will try to fetch a binary delta and apply it.

See also [AppImageSpec, section 'update-information'](https://github.com/AppImage/AppImageSpec/blob/master/draft.md#update-information).

Of course, the updateinfo string is an optional feature to have for AppImages. But to me as an AppImage user it is very convenient, because I do not need to manually search for available updates. I can simply run

    appimageupdatetool some.AppImage

and it will check for and apply an update if it is available. If indeed the update was successful, the old AppImage isn't gone for me: now I just have two versions of the AppImage, and I can keep both or delete the one I don't like any more.

**Comment by KurtPfeifle** [comment]
Nice:

```
$>  ./qpdf-8.0.rc1-x86_64.AppImage --appimage-signature

-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1

iQIcBAABAgAGBQJaksZUAAoJEIp10QmYASx+R3kP/0GbwbaxX7NdiAxDbHkGjhGa
CdSJQN6udD2c5TMGm5YPYpjMcPEHfinQzardI5uwKrYSzNRa9+7KbCbT7Nf65b3f
GFx1oLamfEZ98A5uPrhJyIMgk1/9b4KQdpTzgTQoPUiAi10KwKCvhbHr83Ud46Di
YYKm1MPgDsmQp+om+BRXR87S7q1SI1SwscbJ/44aK8u83N+NiijmW//vEXvsEJ/k
AEFipFu4yVJsqfbR+gr4sIxTFR+rYMwlW/OebnQfH2ul8mU3jGFh7dGr2vQyDQM2
nV7l0mHNWg6cVk3SNJdIn5YKvJqEgJ+4K14hNlVQl8XIuTwAN3+mgicczdiOtUcH
aKetFsdCfiJMS6Ka9xuSlEn00zOcMyDe9vQ4fkZWD57YivC/b5eVLoNp0pRR8hjV
4feDcyw005X7EODlWIJPDV4Wzwzkt/pEwTxswdl9k7W1Arcvu3dL4hBqe8hK1cek
Q4agry55CbuUKx6Fx5zII8Odkcw5/0IwbOd0gBoPHBoqcjmf2fXXtrlBJr0SY7rt
efjOsbeCZzhbB4k8ROxJE7uOrDLtHWTPCE7jQQAajIdQw+KMkpqAmUw0mMrBY6qw
8UJbuu2/dnSOdN1GoHp3MwcKGsYsmx6EnBxrdqVsIRezDdzmql2P28528nj3iANq
qDC2SNuPNRT8V18ahnzl
=1kTC
-----END PGP SIGNATURE-----
```

**Comment by jberkenbilt** [comment]
I've fixed busybox-static and added update info. I'm going to do an rc2 and rc3 release to test the update stuff.

**Comment by jberkenbilt** [comment]
@KurtPfeifle I think this is ready to go. I created a fresh user on my system and was able to grab the 8.0.rc2 qpdf AppImage and then update it using AppImageUpdate to 8.0.rc3. The first time around, I got errors about the icons being a non-standard size, so I added code in build-appimage to create square icons, and now everything looks good. I will prepare 8.0.0 later today. Thanks for all your help. If you want to give it one last look, you can take a look at 8.0.rc3. That is, I expect, identical to 8.0 except the version numbers.

**Comment by jberkenbilt** [comment]
Wait, my travis builds are not right. I'll fix that first.

**Comment by jberkenbilt** [comment]
Travis builds are fine now. I'm not supporting update info for travis builds for now.

**Pull request merged**